Rework - Sensors build test

galak opened the pull request.

galak committed:
drivers: sensor: icm42605: Fix build issues

If CONFIG_ICM42605_TRIGGER is not set the driver doesn't build.  There
are a few places that need ifdefs based on CONFIG_ICM42605_TRIGGER for
the driver to build correctly.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
tests: drivers: build_all: sensor: build only a native_posix

The driver build_all tests only need to build a driver once and
so limit the sensor tests to only build on native_posix.

We tweak CONFIG_SYS_CLOCK_TICKS_PER_SEC as part of this change as
hp206c requires this setting to build.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
tests: drivers: build_all: sensor: simplify configs

Now that we only build on native_posix, we can build all the sensors
together in one pass.  This makes it easier to manage the conf files
going forward.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
This is first step in getting improved coverage on the sensors.  Want to get this merged, than will look at getting ALL sensor drivers actually building as part of this test.

nashif commented on tests/drivers/build_all/sensor/prj.conf:
```diff
@@ -1,2 +1,86 @@
 CONFIG_TEST=y
```
where can I buy this board with so many sensors? :-D

galak committed:
drivers: sensor: max17055: Fix compile error with clang

Since we are writing a register it makes more sense for the type
to be unsigned.  This hopefully address a compile warning we get
with clang:

error: implicit conversion from 'int' to 'int16_t' (aka 'short')
changes value from 32768 to -32768 [-Werror,-Wconstant-conversion]

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
drivers: sensor: iis2iclx: Fix compile warning

Fix compile warning with clang and extraneous parentheses

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

avisconti commented:
> This is first step in getting improved coverage on the sensors. Want to get this merged, than will look at getting ALL sensor drivers actually building as part of this test.

I'll take a closer look tomorrow (Monday). It looks somehow interesting...

avisconti reviewed (approved):
nice idea. Not complete yet. I approve to let you proceed further, as you suggested.

avisconti commented:
I already approved, but what happens if we merge it like this?

MaureenHelm commented:
> I already approved, but what happens if we merge it like this?

I don't understand your question. Are you ok with this getting merged?

avisconti commented:
> > I already approved, but what happens if we merge it like this?
> 
> I don't understand your question. Are you ok with this getting merged?

I am, but I would like @galak to comment more on this statement:

``
This is first step in getting improved coverage on the sensors. Want to get this merged, than will look at getting ALL sensor drivers actually building as part of this test. ``

Does it mean that after merging the CI will fail until job will be completed? I think it is just my misunderstanding, that's why I still approve. But maybe some more clarification is required here.

galak commented on drivers/sensor/iis2iclx/iis2iclx_trigger.c:
```diff
@@ -141,9 +141,9 @@ static void iis2iclx_handle_interrupt(const struct device *dev)
-		if ((status.xlda == 0)
+		if (status.xlda == 0
 #if defined(CONFIG_IIS2ICLX_ENABLE_TEMP)
-					&& (status.tda == 0)
```
Yes, there was a warning/error in clang with this.

galak commented:
> Does it mean that after merging the CI will fail until job will be completed? I think it is just my misunderstanding, that's why I still approve. But maybe some more clarification is required here.

CI should still pass with this change.  My comment was that I wanted this merged before making a number of changes on top of this as I expect we'll starting see more failures with sensors not building in different configs.

The follow PR(s) will do:
* Add all current sensors to build all - will start with `sensors.build`
* Add configs for the 3 common configs for trigger mode - None, global thread, and own thread.
